From: Sebastian Keller <skeller@gnome.org>
Date: Wed, 17 Jan 2024 17:45:24 +0100
Subject: backends: Update cursor names to match updated metaphors

adwaita-icon-theme updated its cursor metaphors and changed all DnD
cursors to use arrows instead of hands, except for the grab related
ones. Mutter was using "grabbing" as default DnD cursor, which now
does not match the other DnD cursors ("copy" and "no-drop") anymore.
Change this to the "default" cursor.

Additionally, because the "no-drop" cursor now puts a stronger emphasis
on the crossed out symbol also prefer "default" for
META_CURSOR_DND_IN_DRAG and only use "no-drop" for things that
explicitly don't accept a drop.

Related: https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/merge_requests/63
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3532>
(cherry picked from commit f99c6badb95309e47e019d3622c9db5dd559da30)

Origin: 46.beta
---
 src/backends/meta-cursor-sprite-xcursor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/backends/meta-cursor-sprite-xcursor.c b/src/backends/meta-cursor-sprite-xcursor.c
index 3b78c16..56b616b 100644
--- a/src/backends/meta-cursor-sprite-xcursor.c
+++ b/src/backends/meta-cursor-sprite-xcursor.c
@@ -72,9 +72,9 @@ translate_meta_cursor (MetaCursor cursor)
     case META_CURSOR_BUSY:
       return "wait";
     case META_CURSOR_DND_IN_DRAG:
-      return "no-drop";
+      return "default";
     case META_CURSOR_DND_MOVE:
-      return "grabbing";
+      return "default";
     case META_CURSOR_DND_COPY:
       return "copy";
     case META_CURSOR_DND_UNSUPPORTED_TARGET:
